utils/build-script-impl

```diff
@@ -1154,15 +1161,8 @@ if [[ "${EXPORT_COMPILE_COMMANDS}" ]] ; then
     )
 fi
 
-if [[ "${EXTRA_CMAKE_OPTIONS}" ]] ; then
-    for cmake_opt in $(echo "${EXTRA_CMAKE_OPTIONS}" | tr "," "\n")
-    do
-        COMMON_CMAKE_OPTIONS=(
-            "${COMMON_CMAKE_OPTIONS[@]}"
-            "${cmake_opt}"
-        )
-    done
-fi
+# Convert to an array.
+eval EXTRA_CMAKE_OPTIONS=(${EXTRA_CMAKE_OPTIONS})
```

Comment:
Because `$EXTRA_CMAKE_OPTIONS` from `build-script` is *always quoted* strings separated by spaces. It's safe to eval, and as far as I know, `eval` is the only way to expand them.
`./build-script --extra-cmake-options=-DMY_OPT=\$FOO` or `./build-script --extra-cmake-options=-DMY_OPT='$FOO'` works as expected.